### DISCUSSION TITLE
[Feature] Support Nikon High-Efficiency RAW and High-Efficiency RAW*

### DISCUSSION THREAD

**Discussion by odedia** [immich-app/immich#7826] [Feature Request]
### The feature

In the latest Nikon mirrorless models (Z8, Z9, ZF), they introduced a new compressed RAW format that greatly reduces the file size but mathematically making it indistinguishable from lossless compressed RAW.
Would be great if it would be possible to add support for such RAW files to immich.
Do note - Lightroom supports this format, but Apple Photos does not. It does allow you to store the RAW files since it's just a NEF file, but it would not display it as of today. I usually use a RAW+JPEG combo, and that works great.
I believe the format itself is based on TICO-RAW which is an open spec.
More info here: https://www.intopix.com/tico-raw
And here: https://www.nikonimgsupport.com/na/NSG_article?articleNo=000051981&lang=en_SG

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
We use [libraw](https://github.com/LibRaw/LibRaw) for processing RAW files, so it comes down to them supporting this format. Feel free to make an issue for it there if there isn't one already.

**Comment by kcossabo** [comment]
According to 

https://github.com/LibRaw/LibRaw/issues/642

there is an embedded jpeg that could be used?

**Reply by mertalev** [reply]
You can enable the option to prefer embedded previews in the Image Settings if you'd like to go down that route.

**Comment by gr3enarr0w** [comment]
Is this still not working 
This would be a blocker for me
Per the Libraw ticket it’s supported

**Comment by timpatten** [comment]
Running into this issue as well.  Not seeing previews, nor the actual image itself, in Immich. Using Nikon Z8 RAW NEF files and nothing shows.  I don't see any issues on the LibRaw pages regarding this.  

If it helps, i'm using external libraries for all of my files and not doing the upload through Immich itself.  

Any updates?

**Reply by mertalev** [reply]
We're using the absolute bleeding edge of libraw right now, so if it doesn't work then there's not much else for us to do here.

**Reply by kcossabo** [reply]
Thank You for the work. 

But, I opened a case with the libraw team too, and Nikon, ya that small Camera company is not enough of a focus yet.

Read all the great reviews, and would love to use the product, but Nikon Z8 is the top camera, and it is not supported. Not pointing fingers, just as a user, I will monitor progress, but as a photographer looking for a solution for a professional camera libraw is just a toy.

**Reply by mertalev** [reply]
Sorry, there'll always be gaps in what a given RAW library supports. You can use a different tool to convert those files to DNG before importing them to Immich. Alternatively, you can enable the "Prefer embedded preview" setting in Immich (assuming the raws have embedded previews).

**Reply by kcossabo** [reply]
Thank You - AGREE 100%, but May 2023 Z8 was launched. 

Nothing against your product, everything I read is it is top notch, and I played with it with my iPhone and older RAW photos, it is good, actually amazing. 

I will play with my install for the ‘prefer embedded’ to see what that is like, but I am not converting 1000’s of photos to DNG. That would work, just not a pro workflow.

Not sure what ‘pressure’ a downstream consumer of a library can give, but, this ‘issue’ keeps me with Lightroom and Adobe (who also is late to adopt, but in months not years).

keep up the great work, hope the library catches up.

**Reply by StackOvenFlow** [reply]
I have a workaround in place on my docker instance right now, documented/explained what's and why's on another issue that was closed as duplicate of this one, but here we go: https://github.com/immich-app/immich/issues/19139

**Reply by mertalev** [reply]
There is an image setting to extract and use embedded previews. Is there a particular reason you patched this into immich yourself?

**Reply by StackOvenFlow** [reply]
Honestly I didn't read the doc nor checked the Admin options before (started using Immich a couple days ago).
I was redoing a bunch of my homelab and just went straight to the code, ended up updating the logic to also use exif for image orientation.
My bad 😥

**Reply by johnnysbug** [reply]
Does anyone know what Synology Photos uses to process NEF images? It seems to display NEF images fine, but the same image viewed in Immich is blurry. However when first rendered, it looks crisp, but after a second or two it gets replaced with a blurry version.

**Comment by johnnysbug** [comment]
For me, this appears to only be an issue on the iOS mobile app. I've attached a screen recording from my iPhone to demonstrate what I see. When panning to the last image, it looks fine initially, but after a few seconds, the image blurs and shows a black border at the top and bottom of the image. Viewing this same image on the web (chromium browser) it renders fine.

Additionally, if I try to download this image via the iOS app, it downloads in the blurred state.

https://github.com/user-attachments/assets/022b000b-a0e2-4115-b047-5c3ce8972cf7

**Comment by kcossabo** [comment]
Jan 30, 2026 - I thought this was resolved in the 2.5.0 but it looks like it still has issues??

Will this ever be fixed?

**Reply by johnnysbug** [reply]
I got around the blurry images by disabling these properties:
<img width="1179" height="720" alt="image" src="https://github.com/user-attachments/assets/da30ff95-7348-4347-9d31-d7f7096bda54" />
But it is just a workaround

**Comment by vfurnes** [comment]
Still experiencing this issue on v2.7.4 with Nikon Z8 NEF files using an external library (read-only mount).

Thumbnail generation reports as successful in the Jobs panel, but no thumbnail file is created on disk and the image displays "Error loading image" in the web UI.

From the server logs:
```
Error: Input file has corrupt header: magickload: Magick: Unsupported file format or not RAW file `/usr/src/app/external/Image Folder/DSC_XXXX.NEF' @ error/dng.c/ReadDNGImage/536 (null)
    at Sharp.toBuffer (/usr/src/app/server/node_modules/.pnpm/sharp@0.34.5/node_modules/sharp/lib/output.js:163:17)
```

`magickload` is attempting to read the Z8 NEF files as DNG format and failing. This affects all NEF files from the Z8 across multiple shooting dates. Other RAW formats (Sony ARW) in the same library work fine.

The embedded preview workaround is not ideal. Hoping a fix can be released soon!

**Reply by kikoso** [reply]
I am having the same issue.

**Reply by Northwave** [reply]
I also have this issue. this is really annoying as i sometimes use HE* und sometimes only lossless compression. HE* is what i really need when im birding so having less used space and also better speed with this. Would love to see this working

**Reply by kikoso** [reply]
@Northwave , it seems to be an issue related to LibRaw. This thread is being cross-references [from here](https://github.com/LibRaw/LibRaw/issues/642).

**Reply by Northwave** [reply]
@kikoso sure, ive seen this, but he also stated at the end that thumbnail extraction is working. i would love to have this, instead even this wont work so im not able to sort my pics.

**Comment by zexindai** [comment]
Temporarily solve it by active the "Prefer embedded preview"

**Reply by kikoso** [reply]
This still doesn't work for me.

**Reply by DuckThom** [reply]
@kikoso It worked for me after refreshing the thumbnails. Before, the RAW images looked something like this

<img width="369" height="252" alt="image" src="https://github.com/user-attachments/assets/e369d94c-16d5-4fc4-a538-df05ce6e5f3a" />

After enabling "Prefer embedded preview" and then refreshing the thumbnails, the image appeared correctly for me

<img width="286" height="187" alt="image" src="https://github.com/user-attachments/assets/5ce09fd1-c299-49a9-a894-21ea318fd02b" />

**Reply by kikoso** [reply]
Thanks @DuckThom ! Refreshing the thumbnails made the trick. Now doing it in bulk for all the RAWs :)
